scripts/generate_sudo_list.py

```diff
@@ -0,0 +1,69 @@
+"""Helper script to generate sudo list doc."""
+import argparse
+import filecmp
+import sys
+import tempfile
+from pathlib import Path
+
+import yaml
+
+
+def collect_sudo_commands(path=None):
+    """Collect all ansible cmds that require sudo."""
+    if not path:
+        path = Path("quipucords/scanner/network/runner")
+    list_cmds = []
+    for playbook in path.glob("roles/*/tasks/main.yml"):
+        for task in yaml.safe_load(playbook.open()):
+            if task.get("become", ""):
+                list_cmds.append(task.get("raw", ""))
+    return list_cmds
```

Comment:
Before we return the list, we may want to drop any empty strings. I wonder if there's a task somewhere that put an empty string resulting in that stray blank line in the output. 🤔 

Consider:
```
    return [cmd.strip() for cmd in list_cmds if isinstance(cmd, str) and cmd.strip()]
```

That list comprehension will strip whitespace from each item and return only non-empty items. For example:

```
>>> list_cmds = ['hello', '', '1', 2, 'world', 'world', ' ', '0', 'False', None, False, True]
>>> [cmd.strip() for cmd in list_cmds if isinstance(cmd, str) and cmd.strip()]
['hello', '1', 'world', 'world', '0', 'False']
```

We may _also_ want to sort the final list, though, because `path.glob` **does not guarantee order**. Running the same command at different times or on different machines can result in a different ordered list, even if the actual content would be the same. So, consider this suggestion to unique (via `set`) and sort the list:

```suggestion
    list_cmds = [cmd.strip() for cmd in list_cmds if isinstance(cmd, str) and cmd.strip()]
    list_cmds = sorted(set(list_cmds))
    return list_cmds
```

That _could_ be a one-line return statement, but breaking it into a few lines might help with future debugging if we encounter issues. That's just a subjective opinion, though. :shrug: Anyhow, using the previous example with this new code, it would return:

```
['0', '1', 'False', 'hello', 'world']
```

No empty values, no duplicates, and order is sorted. 🎉

Reply:
good idea! thanks